lib/init.c

```diff
@@ -192,8 +205,11 @@ struct smb2_url *smb2_parse_url(struct smb2_context *smb2, const char *url)
                 smb2_set_error(smb2, "URL is too long");
                 return NULL;
         }
+#if defined(__amigaos4__) || defined(__AMIGA__) || defined(__AROS__)
+        strcpy(str, url + 6);
+#else
         strncpy(str, url + 6, MAX_URL_SIZE);
-
+#endif
```

Comment:
Add a 
#if defined(__amigaos4__) || defined(__AMIGA__) || defined(__AROS__)
#define strncpy(a,b,c) strcpy(a,b)
#endif

to compat.h

Reply:
Added here:
https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa